tests/python/pants_test/backend/python/tasks/test_python_run_integration.py

```diff
@@ -39,17 +69,16 @@ def test_run_27_and_then_3(self):
       pants_run_3 = self.run_pants(
         command=['run', '{}:echo_interpreter_version_3'.format(self.testproject),
                  '--python-setup-interpreter-constraints=CPython>=2.7,<3',
-                 '--python-setup-interpreter-constraints=CPython>=3.3'],
+                 '--python-setup-interpreter-constraints=CPython>=3.4,<3.7'],
```

Comment:
Might be worth constantizing `CPython>=3.4,<3.7` and possibly some of the other ranges.

Reply:
Agreed - they are slapdash right now scattered everywhere and they'll all need to change when we bump our floor from 3.4 to - hopefully - 3.6.  I've filed #6458 to follow-up.

Reply:
And https://github.com/pantsbuild/pants/pull/6446/files#diff-bbf10c952a12c5ffb5da4f44a2763b9f will provide a home for those ranges.